client/my-sites/hosting/cache-card/index.js

```diff
@@ -72,24 +77,30 @@ export const CacheCard = ( {
 		isInitialLoading: getEdgeCacheInitialLoading,
 	} = useEdgeCacheQuery( siteId );
 
+	const dispatch = useDispatch();
+
 	const isEdgeCacheEligible = ! isPrivate && ! isComingSoon;
 
-	const { toggleEdgeCache, isLoading: toggleEdgeCacheLoading } = useToggleEdgeCacheMutation(
-		siteId,
-		{
-			onSettled: ( ...args ) => {
-				const active = args[ 2 ];
-				recordTracksEvent(
-					active
-						? 'calypso_hosting_configuration_edge_cache_enable'
-						: 'calypso_hosting_configuration_edge_cache_disable',
-					{
-						site_id: siteId,
-					}
-				);
-			},
-		}
-	);
+	const { setEdgeCache, isLoading: isEdgeCacheMutating } = useSetEdgeCacheMutation( {
```

Comment:
> the special `isLoading` handling doesn't appear to be used anywhere

Oops, special `isLoading` behaviour was being used and I just missed it. The built in `isLoading` behaviour is a little dumber and only returns true for this specific mutation on this specific mount of the component. So it doesn't handle page changes. The network request is usually pretty quick so it's hard to test this, but in the spirit of not changing the behaviour I'll push a change. I think it's in the sprit of this PR anyway.

Reply:
036f4b6ce50d9be44e2c1f83f25ac5d069435a31